observer_test.go

```diff
@@ -0,0 +1,95 @@
+package jaeger
+
+import (
+	"testing"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEmptyObserver(t *testing.T) {
+	tracer, closer := NewTracer("test", NewConstSampler(true), NewInMemoryReporter())
+	defer closer.Close()
+	s := tracer.StartSpan("test", ext.RPCServerOption(nil))
+	s.Finish()
+	assert.Equal(t, s.(*span).observer, noopSpanObserver)
+}
+
+func TestObservers(t *testing.T) {
+	tracer, closer := NewTracer(
+		"test",
+		NewConstSampler(true),
+		NewInMemoryReporter(),
+		TracerOptions.Observer(testObserver{}),
+		TracerOptions.Observer(testObserver{}),
+	)
+	defer closer.Close()
+
+	s := tracer.StartSpan("test", ext.RPCServerOption(nil))
+
+	forEachObs := func(f func(so *testSpanObserver)) {
+		observers := s.(*span).observer.(spanObserver).observers
+		assert.Len(t, observers, 2)
+		for _, so := range observers {
+			f(so.(*testSpanObserver))
+		}
+	}
+
+	forEachObs(func(so *testSpanObserver) {
+		assert.Equal(t, testSpanObserver{
+			operationName: "test",
+			tags: map[string]interface{}{
+				"span.kind": ext.SpanKindRPCServerEnum,
+			},
+		}, *so)
+	})
+
+	s.SetOperationName("test2")
+	s.SetTag("bender", "rodriguez")
+	forEachObs(func(so *testSpanObserver) {
+		assert.Equal(t, testSpanObserver{
+			operationName: "test2",
+			tags: map[string]interface{}{
+				"span.kind": ext.SpanKindRPCServerEnum,
+				"bender":    "rodriguez",
+			},
+		}, *so)
+	})
+
+	s.Finish()
+	forEachObs(func(so *testSpanObserver) {
+		assert.True(t, so.finished)
+	})
+}
+
+type testObserver struct{}
+
+type testSpanObserver struct {
```

Comment:
for the clients, we want them as lightweight as possible, the fewer deps the better